packages/common/src/extensions/slickRowBasedEdit.ts

```diff
@@ -0,0 +1,596 @@
+          className:
+            `${
+              options.rowBasedEditOptions?.actionButtons?.editButtonClassName ||
+              'button-style padding-1px mr-2'
+            } action-btns ` + BTN_ACTION_EDIT,
+          title:
+            options.rowBasedEditOptions?.actionButtons?.editButtonTitle ||
+            'Edit the Row',
+          style: { display: isInEditMode ? 'none' : '' },
+        })
+      )
+      .appendChild(
+        createDomElement('span', {
+          className:
+            options.rowBasedEditOptions?.actionButtons
+              ?.iconEditButtonClassName || 'mdi mdi-table-edit color-primary',
+        })
+      );
+    actionFragment
+      .appendChild(
+        createDomElement('span', {
+          className:
+            `${
+              options.rowBasedEditOptions?.actionButtons
+                ?.deleteButtonClassName || 'button-style padding-1px'
+            } action-btns ` + BTN_ACTION_DELETE,
+          title:
+            options.rowBasedEditOptions?.actionButtons?.deleteButtonTitle ||
+            'Delete the Row',
+          style: { display: isInEditMode ? 'none' : '' },
+        })
+      )
+      .appendChild(
+        createDomElement('span', {
+          className:
+            options.rowBasedEditOptions?.actionButtons
+              ?.iconDeleteButtonClassName || 'mdi mdi-close color-danger',
+        })
+      );
+    actionFragment
+      .appendChild(
+        createDomElement('span', {
+          className:
+            `${
+              options.rowBasedEditOptions?.actionButtons
+                ?.updateButtonClassName || 'button-style padding-1px mr-2'
+            } action-btns ` + BTN_ACTION_UPDATE,
+          title:
+            options.rowBasedEditOptions?.actionButtons?.updateButtonTitle ||
+            'Update the Row',
+          style: { display: !isInEditMode ? 'none' : '' },
+        })
+      )
+      .appendChild(
+        createDomElement('span', {
+          className:
+            options.rowBasedEditOptions?.actionButtons
+              ?.iconUpdateButtonClassName || 'mdi mdi-check-bold color-success',
+        })
+      );
+    actionFragment
+      .appendChild(
+        createDomElement('span', {
+          className:
+            `${
+              options.rowBasedEditOptions?.actionButtons
+                ?.cancelButtonClassName || 'button-style padding-1px'
+            } action-btns ` + BTN_ACTION_CANCEL,
+          title:
+            options.rowBasedEditOptions?.actionButtons?.cancelButtonTitle ||
+            'Cancel changes of the Row',
+          style: { display: !isInEditMode ? 'none' : '' },
+        })
+      )
+      .appendChild(
+        createDomElement('span', {
+          className:
+            options.rowBasedEditOptions?.actionButtons
+              ?.iconCancelButtonClassName || 'mdi mdi-cancel color-danger',
+        })
+      );
+
+    return actionFragment;
+  }
+
+  protected onBeforeEditCellHandler = (
+    _e: Event,
```

Comment:
```suggestion
    _e: SlickEventData,
```
because of recent PR #1327